### ISSUE TITLE
[Question] Is there a box-wide legality check? 

### ISSUE THREAD

**Issue by greenDarkness** [kwsch/PKHeX#739]
Sometimes when I open PKHeX it marks all my Pokemon as illegal, even the legal ones. If I few the Pokemon, and then click the red caution icon at the top it then gives me a legal notification. If I view the Pokemon again then it goes back to illegal. Viewing, and then setting the Pokemon fixes this issue. 

Below is a PK7 of an afflicted Pokemon.
http://www.mediafire.com/file/u3z1ooaobx7dmvx/132_%E2%98%85_-_Ditto_-_EE89382648C6.pk7

Thanks for reading!

**Comment by kwsch** [comment]
database

When saving a PKM from tabs, PKHeX fixes common user errors and illegal values. Loading something to tabs and re-checking the legality causes PKHeX to save the PKM, fixing the illegal property.

Refer to https://github.com/kwsch/PKHeX/issues/604 item 3.

**Comment by greenDarkness** [comment]
Is there a way to do that without loading each Pokemon into a tab and then checking the legality? 408 transferred Pokemon are effected by this, and it would take quite some time to load and save each one up manually. 

Thank you for answering my question.

**Comment by kwsch** [comment]
that's the purpose of the batch editor

=HT_Gender=2
.HT_Gender=0

**Comment by greenDarkness** [comment]
It says that it didn't modify any files. And they are all still marked as illegal. 
Alert
Modified 0/585 files.

**Comment by kwsch** [comment]
![image](https://cloud.githubusercontent.com/assets/6393368/22282481/94ab7138-e291-11e6-9a60-0e86b14e9d07.png)

**Comment by greenDarkness** [comment]
https://snag.gy/JK75Uv.jpg

**Comment by kwsch** [comment]
Try setting a breakpoint [here](https://github.com/kwsch/PKHeX/blob/2cff51be61528bd62dec8b126507f5dce5e615f1/PKHeX/Legality/Analysis.cs#L49) and inspect what the Parse list contains in terms of legality analysis.

Loading a PKM should trigger it the first time, which should give a clue what is the actual error that is being automatically fixed => replicate in batch editor.

**Comment by greenDarkness** [comment]
I'm not quite sure how to go about inserting a breakpoint, as currently I'm code illiterate. Would it be possible to dumb it down for me? Thanks so much for taking time to look at this!
 I opened up an illegal, and a fixed version of that illegal in my text editor and the 10th line is 0000 as opposed to  1d31 on the illegal mon, and that's the only difference. Does that help any?
https://snag.gy/McAHVo.jpg

**Comment by kwsch** [comment]
Previous commits have added the ability to legality check a PKM in a box without loading to tabs.

**Comment by PKMWM1** [comment]
How do I do a legality analysis? Its not an option anywhere in any of the tabs or settings. Clicking the PKM Database option takes me to a screen that has arbitrary search options but not the legality analysis or the database =>tools => Search settings as stated above.

**Comment by kwsch** [comment]
Search for everything that is illegal (un-tick Legal) and inspect away. If it's a tricky one that flips to Legal (auto-fixed), the location that the database loaded the file from is shown in the bottom left corner.
